app/core/src/main/java/com/fsck/k9/cache/ExpiringCache.kt

```diff
@@ -0,0 +1,43 @@
+package com.fsck.k9.cache
+
+internal class ExpiringCache<KEY : Any, VALUE : Any>(
```

Comment:
This caching strategy might be too simple. We'll have to check if there's still a noticeable benefit to having such a cache.

Reply:
The benefit is little, but it allows me to break up some dependencies without changing the current behavior too much. Also it is easily replaceable by a more sofisticated strategy when needed, e.g LRU cache ...